lib/jekyll/excerpt.rb

```diff
@@ -158,6 +158,11 @@ def extract_excerpt(doc_content)
 
     def liquid_block?(tag_name)
       Liquid::Template.tags[tag_name].superclass == Liquid::Block
+    rescue NoMethodError
+      Jekyll.logger.error "Error:",
+        "Excerpt in #{doc.relative_path} appears to contain a Liquid tag which " \
+        "couldn't be parsed."
+      raise
```

Comment:
This seems like a nice compromise to me for something like a `3.8.2`: a much more helpful error message without any material change in behavior :+1:

Reply:
Can we shorten the message a bit..? :
```ruby
Jekyll.logger.error "Error:",
  "Liquid tag in #{doc.relative_path} couldn't be parsed."
```

Reply:
I would favor a more descriptive error message over a less descriptive one, but if you think this is sufficient, :+1:

Reply:
This is the reason why we decided to add a helpful error message.. :wink: